tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -177,6 +177,45 @@
 
 #define DEV_NULL_STR    0
 
+#define KERNEL_CONFIG_BPF                       1
+#define KERNEL_CONFIG_BPF_SYSCALL               2
+#define KERNEL_CONFIG_HAVE_EBPF_JIT             3
+#define KERNEL_CONFIG_BPF_JIT                   4
+#define KERNEL_CONFIG_BPF_JIT_ALWAYS_ON         5
+#define KERNEL_CONFIG_CGROUPS                   6
+#define KERNEL_CONFIG_CGROUP_BPF                7
+#define KERNEL_CONFIG_CGROUP_NET_CLASSID        8
+#define KERNEL_CONFIG_SOCK_CGROUP_DATA          9
+#define KERNEL_CONFIG_BPF_EVENTS                10
+#define KERNEL_CONFIG_KPROBE_EVENTS             11
+#define KERNEL_CONFIG_UPROBE_EVENTS             12
+#define KERNEL_CONFIG_TRACING                   13
+#define KERNEL_CONFIG_FTRACE_SYSCALLS           14
+#define KERNEL_CONFIG_FUNCTION_ERROR_INJECTION  15
+#define KERNEL_CONFIG_BPF_KPROBE_OVERRIDE       16
+#define KERNEL_CONFIG_NET                       17
+#define KERNEL_CONFIG_XDP_SOCKETS               18
+#define KERNEL_CONFIG_LWTUNNEL_BPF              19
+#define KERNEL_CONFIG_NET_ACT_BPF               20
+#define KERNEL_CONFIG_NET_CLS_BPF               21
+#define KERNEL_CONFIG_NET_CLS_ACT               22
+#define KERNEL_CONFIG_NET_SCH_INGRESS           23
+#define KERNEL_CONFIG_XFRM                      24
+#define KERNEL_CONFIG_IP_ROUTE_CLASSID          25
+#define KERNEL_CONFIG_IPV6_SEG6_BPF             26
+#define KERNEL_CONFIG_BPF_LIRC_MODE2            27
+#define KERNEL_CONFIG_BPF_STREAM_PARSER         28
+#define KERNEL_CONFIG_NETFILTER_XT_MATCH_BPF    29
+#define KERNEL_CONFIG_BPFILTER                  30
+#define KERNEL_CONFIG_BPFILTER_UMH              31
+#define KERNEL_CONFIG_TEST_BPF                  32
+#define KERNEL_CONFIG_HZ                        33
+#define KERNEL_CONFIG_DEBUG_INFO_BTF            34
+#define KERNEL_CONFIG_DEBUG_INFO_BTF_MODULES    35
+#define KERNEL_CONFIG_BPF_LSM                   36
+#define KERNEL_CONFIG_BPF_PRELOAD               37
+#define KERNEL_CONFIG_BPF_PRELOAD_UMD           38
```

Comment:
I think we can leave this out for now, and add new defines when needed

Reply:
I think it's warranted to have them in libbpfgo, and since the enum values are already set there, they should exist here.

Reply:
Why should they exist here if they are not being used?

Reply:
To reserve the numbers since they must correspond to the ones in libbpfgo anyway.

Reply:
But why should we reserve them? They will never be used in the BPF context. It is just a "define" directive which can be added when needed

Reply:
My thinking is that we want to add any possible constants to libbpfgo because people might want to use them even if tracee-ebpf doesn't. Since the constants are already defined in libbpfgo I felt that it would be a good idea to just have constants defined for simplifying future development. 

However upon thinking about it I realize that this isn't even the proper way of handling feature flagging in BPF code. In the case of CO:RE stuff, I'm using a conditional compile flag. I'm removing this from the bpf code altogether.